### PR TITLE
Http module improvements

### PR DESCRIPTION
**Pull request by ruflin**
This is a follow up PR for https://github.com/elastic/beats/pull/4156

* Add environment with small golang web service
* Remove not needed body fields
* Add defaults to configs
* Add system tests
* Update config with all config options

### CODE REVIEW

**Comment by ruflin** [comment]
jenkins, retest it

File: metricbeat/docs/modules/http.asciidoc [hunk 1]
```diff
@@ -5,8 +5,8 @@ This file is generated! See scripts/docs_collector.py
 [[metricbeat-module-http]]
 == http Module
 
-Http module is a [Metricbeat](https://www.elastic.co/products/beats/metricbeat) used to call arbitrary HTTP endpoints for which not a dedicated metricbeat is available.
-Multiple endpoints can be configured which are polled in a regular interval and the result is shipped to the configured output channel.
+Http module is a [Metricbeat](https://www.elastic.co/products/beats/metricbeat) used to call arbitrary HTTP endpoints for which not a dedicated metricbeat module is available.
+Multiple endpoints can be configured which are polled in a regular interval and the result is shipped to the configured output channel. It is recommend to fetch install a metricbeat instance on each host from which data should be fetched.
```
Comment (tsg) [inline metricbeat/docs/modules/http.asciidoc]: Some typos in this phrase: "It is recommend**ed** to (fetch) install..."

File: metricbeat/docs/modules/http.asciidoc [hunk 2]
```diff
@@ -5,8 +5,8 @@ This file is generated! See scripts/docs_collector.py
 [[metricbeat-module-http]]
 == http Module
 
-Http module is a [Metricbeat](https://www.elastic.co/products/beats/metricbeat) used to call arbitrary HTTP endpoints for which not a dedicated metricbeat is available.
-Multiple endpoints can be configured which are polled in a regular interval and the result is shipped to the configured output channel.
+Http module is a [Metricbeat](https://www.elastic.co/products/beats/metricbeat) used to call arbitrary HTTP endpoints for which not a dedicated metricbeat module is available.
```
Comment (tsg) [inline metricbeat/docs/modules/http.asciidoc]: I think there's a missing "module" after Metricbeat, and an extra "not".

File: metricbeat/docs/modules/http.asciidoc [hunk 3]
```diff
@@ -5,8 +5,8 @@ This file is generated! See scripts/docs_collector.py
 [[metricbeat-module-http]]
 == http Module
```
Comment (tsg) [inline metricbeat/docs/modules/http.asciidoc]: "http Module" looks weird in the docs due to capitalization. I think there was a way to define the title in `fields.yml`?

File: metricbeat/docs/modules/http.asciidoc (hunk 1 above)
Comment (ruflin) [inline metricbeat/docs/modules/http.asciidoc]: fixed

File: metricbeat/docs/modules/http.asciidoc (hunk 2 above)
Comment (ruflin) [inline metricbeat/docs/modules/http.asciidoc]: changed to `module used to call arbitrary HTTP endpoints for which a dedicated metricbeat module is not available.`

File: metricbeat/docs/modules/http.asciidoc (hunk 3 above)
Comment (ruflin) [inline metricbeat/docs/modules/http.asciidoc]: changed the title in `fields.yml` to `HTTP`.

**Comment by ruflin** [comment]
Review comments applied and pushed.

**Comment by ruflin** [comment]
jenkins, retest it

File: metricbeat/module/http/_meta/docs.asciidoc [hunk 4]
```diff
@@ -1,7 +1,7 @@
 == http Module
 
-Http module is a [Metricbeat](https://www.elastic.co/products/beats/metricbeat) used to call arbitrary HTTP endpoints for which not a dedicated metricbeat is available.
-Multiple endpoints can be configured which are polled in a regular interval and the result is shipped to the configured output channel.
+Http module is a [Metricbeat](https://www.elastic.co/products/beats/metricbeat) used to call arbitrary HTTP endpoints for which not a dedicated metricbeat module is available.
+Multiple endpoints can be configured which are polled in a regular interval and the result is shipped to the configured output channel. It is recommend to fetch install a metricbeat instance on each host from which data should be fetched.
```
Comment (monicasarbu) [inline metricbeat/module/http/_meta/docs.asciidoc]: It doesn't sound good to me "to fetch install", maybe just "to install".

Comment (monicasarbu) [inline metricbeat/module/http/_meta/docs.asciidoc]: I noticed that Tudor left the same comment above.

Comment (ruflin) [inline metricbeat/module/http/_meta/docs.asciidoc]: Argh, I didn't push the change.

File: metricbeat/docs/modules/http.asciidoc (hunk 3 above)
Comment (monicasarbu) [inline metricbeat/docs/modules/http.asciidoc]: I would use `HTTP` for the protocol, and `http` for the module name. What about `http module`?

Comment (monicasarbu) [inline metricbeat/docs/modules/http.asciidoc]: Ah, sorry. I just noticed that you changed already to `Http module`. That's perfect.

File: metricbeat/docs/modules/http.asciidoc (hunk 1 above)
Comment (monicasarbu) [inline metricbeat/docs/modules/http.asciidoc]: @ruflin It still displays the old format "It is recommend to fetch install".

**Pull request merged**